assembl/graphql/vote_session.py

```diff
@@ -56,16 +72,20 @@ def mutate(root, args, context, info):
         if discussion_phase is None:
             raise Exception(
                 "A vote session requires a discussion phase, check discussionPhaseId value")
-        phase_slug = "voteSession"
-        if discussion_phase.identifier != phase_slug:
+        phase_identifier = "voteSession"
+        if discussion_phase.identifier != phase_identifier:
             raise Exception(
-                "A vote session can only be created or edited with a '{}' discussion phase, check discussionPhaseId value".format(phase_slug))
+                "A vote session can only be created or edited with a '{}' discussion phase, check discussionPhaseId value".format(phase_identifier))
+
+        require_permission = make_permissions_querier(models.VoteSession, context)
 
         # TODO: see if we can avoid this next(iter( thing with a one-to-one relationship
         vote_session = next(iter(discussion_phase.vote_session or []), None)
         if vote_session is None:
-            vote_session = models.VoteSession(
-                discussion_phase=discussion_phase)
+            require_permission(CrudPermissions.CREATE)
+            vote_session = models.VoteSession(discussion_phase=discussion_phase)
+        else:
+            require_permission(CrudPermissions.UPDATE)
```

Comment:
This is wrong, CrudPermissions.UPDATE needs to be checked on the object you update, so with `obj.user_can` api. `cls.user_can_cls` is only used for `CrudPermissions.CREATE`

Reply:
I was wondering about that, thanks

Reply:
After writing tests it seems that the class version is fine for this case.

Reply:
Yes because we don't really care here about owner (is_owner method), this is the difference between user_can_cls and user_can which check is_owner.

Reply:
But you really should support the right api now, this is important if another developer uses your require_permission api.

Reply:
On it